Add runbook (#28)
Financial-Times/public-things-api: runbooks/runbook.md

# UPP - Public Things API

> Provides a publicly available API that gives basic information about things in the graph store. It is written in Go, uses Neo4j as a data store, and is deployed in the UPP Kubernetes stack

Reviewer: I would leave only the first sentence here and have a corrected description in the "Architecture" section.

## Code

public-things-api

## Primary URL

> <https://github.com/Financial-Times/public-things-api>

Reviewer: This is a public API, its URL is:
```suggestion
<http://api.ft.com/things>
```

## Service Tier

Bronze

## Lifecycle Stage

Production

## Delivered By

content

## Supported By

content

## Known About By

- elitsa.pavlova
- kalin.arsov
- ivan.nikolov
- marina.chompalova
- miroslav.gatsanoga

## Host Platform

AWS

## Architecture

> Provides a publicly available API that gives basic information about things in the graph store. It is written in Go, uses Neo4j as a data store, and is deployed in the UPP Kubernetes stack

Reviewer: This is not correct for quite some time. My suggestion:
```suggestion
The service calls Public Concepts API to retrieve the information from the Neo4j data store.
It is deployed in the UPP Delivery Kubernetes clusters.
```

## Contains Personal Data

No

## Contains Sensitive Data

No

## Failover Architecture Type

ActiveActive

## Failover Process Type

FullyAutomated

## Failback Process Type

PartiallyAutomated

## Failover Details

> See the [failover guides for more details](https://github.com/Financial-Times/upp-docs/tree/master/failover-guides)

Reviewer: I would prefer to have here the cluster specific failover guide:
```suggestion
The service is deployed in both Delivery clusters. The failover guide for the cluster is located here: https://github.com/Financial-Times/upp-docs/tree/master/failover-guides/delivery-cluster
```

## Data Recovery Process Type

NotApplicable

## Data Recovery Details

The service does not store data, so it does not require any data recovery steps.

## Release Process Type

PartiallyAutomated

## Rollback Process Type

Manual

## Release Details

The release is triggered by making a Github release which is then picked up by a Jenkins multibranch pipeline. The Jenkins pipeline should be manually started in order for it to deploy the helm package to the Kubernetes clusters.

## Key Management Process Type

NotApplicable

## Key Management Details

There is no key rotation procedure for this system.

## Monitoring

> Look for the pods in the cluster health endpoint and click to see pod health and checks:
> 
> - <https://upp-prod-delivery-eu.upp.ft.com/__health>
> - <https://upp-prod-delivery-us.upp.ft.com/__health>

Reviewer: The specific links would be way more useful:
```suggestion
- Delivery-Prod-EU health: <https://upp-prod-delivery-eu.upp.ft.com/__health/__pods-health?service-name=public-things-api>
- Delivery-Prod-US health: <https://upp-prod-delivery-us.upp.ft.com/__health/__pods-health?service-name=public-things-api>
```

# UPP - Public Things API

The service calls Public Concepts API to retrieve the information from the Neo4j data store

## Code

public-things-api

## Primary URL

<http://api.ft.com/things>

## Service Tier

Bronze

## Lifecycle Stage

Production

## Delivered By

content

## Supported By

content

## Known About By

- elitsa.pavlova
- kalin.arsov
- ivan.nikolov
- marina.chompalova
- miroslav.gatsanoga

## Host Platform

AWS

## Architecture

The service calls Public Concepts API to retrieve the information from the Neo4j data store.
It is deployed in the UPP Delivery Kubernetes clusters.

## Contains Personal Data

No

## Contains Sensitive Data

No

> 

Reviewer: ```suggestion

## Dependencies

- public-concepts-api

```
